src/Microsoft.AspNet.Hosting/DefaultRequestIdentifierFeature.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNet.Http;
+
+namespace Microsoft.AspNet.Hosting
+{
+    public class DefaultRequestIdentifierFeature : IRequestIdentifierFeature
+    {
+        public DefaultRequestIdentifierFeature()
+        {
+            TraceIdentifier = Guid.NewGuid();
+        }
+
+        public Guid TraceIdentifier { get; }
```

Comment:
For your FYI information in the future I think this will change from a `Guid` to a `string` but don't worry about that here.